### PR TITLE
Extend automatically generated docs

### LINKED ISSUE #99: Extend the existing automatically generated Swagger docs
With more specific information about the endpoints, what they do and their purpose.

How?

https://fastapi.tiangolo.com/tutorial/metadata/

# Acceptance Criteria

- [ ] Functions listed for each model object
- [ ] App description and metadata entered

### PR DESCRIPTION
**Pull request by BenCapper**
- Add Duffy description
- Document Node, Session, Project and Chassis operations
- Tags for Nodes, Sessions, Projects and Chassis
- Add app metadata

Signed-off-by: Ben Capper <BenCapper19@gmail.com>

### CODE REVIEW

File: duffy/app/main.py [hunk 1]
```diff
@@ -5,7 +5,59 @@
...
+## Chassis
+
+- Get all Chassis: Returns all chassis
+- Create a Chassis: Create a chassis with the specified **name** and an optional **description**
+- Get a Chassis: Return the chassis with the specified **ID**
+- Delete a Chassis: Delete the chassis with the specified **ID**
+
+"""
+
+tags_metadta = [
```
Comment (nphilipp) [inline duffy/app/main.py]: I think this should be `tags_metadata`. 🙂

File: duffy/app/main.py [hunk 2]
```diff
@@ -5,7 +5,59 @@
 from .controllers import chassis, node, project, session
 
 log = logging.getLogger(__name__)
-app = FastAPI()
+
+description = """
+Duffy is the middle layer running ci.centos.org that manages the provisioning,
+maintenance and teardown / rebuild of the Nodes (physical hardware for now, VMs coming soon)
+that are used to run the tests in the CI Cluster.
```
Comment (nphilipp) [inline duffy/app/main.py]: Let's condense this a little and link ci.centos.org, e.g.:

```
Duffy is the middle layer running [`ci.centos.org`](https://ci.centos.org). It provisions, tears
down and rebuilds physical and virtual machines are used to run tests in the CentOS CI Cluster.
```

File: duffy/app/main.py [hunk 3]
```diff
@@ -5,7 +5,59 @@
...
+- Get a Project: Returns the project with the specified **ID**
+- Delete a Project: Deletes the project with the specified **ID**
+
+## Chassis
+
+- Get all Chassis: Returns all chassis
+- Create a Chassis: Create a chassis with the specified **name** and an optional **description**
+- Get a Chassis: Return the chassis with the specified **ID**
+- Delete a Chassis: Delete the chassis with the specified **ID**
+
```
Comment (nphilipp) [inline duffy/app/main.py]: This is redundant and hides the actual API documentation below, let’s remove it.

File: duffy/app/main.py [hunk 4]
```diff
@@ -5,7 +5,59 @@
...
+tags_metadta = [
+    {"name": "nodes", "description": "Operations with physical and virtual nodes"},
+    {"name": "sessions", "description": "Operations with sessions"},
+    {"name": "projects", "description": "Operations with projects"},
+]
+
+app = FastAPI(
+    title="Duffy",
+    description=description,
+    version=2.0,
```
Comment (nphilipp) [inline duffy/app/main.py]: We're at version 3.0, this is defined in `duffy.version.__version__`, i.e. change this 👆 to this:

```
...
from ..version import __version__
...
app = FastAPI(
    ...
    version=__version__,
    ...
)
```

Also, this passes `2.0` as a floating point value which happens to work here but with a different value might be rounded to something like `1.999999` and it won't work for more than two levels, e.g. `3.0.0` – `version` should be a string.

**Review by nphilipp — CHANGES_REQUESTED** [review]
Apart from the inline comments, the commits so far should be squashed into one before we merge the PR.

File: duffy/app/main.py (hunk 4 above)
Comment (gridhead) [reply duffy/app/main.py]: Addressed here https://github.com/CentOS/duffy/pull/133/commits/1c52ef02660e7638be331c5362d113ebb4e6fa8b

File: duffy/app/main.py (hunk 3 above)
Comment (gridhead) [reply duffy/app/main.py]: Addressed here https://github.com/CentOS/duffy/pull/133/commits/1c52ef02660e7638be331c5362d113ebb4e6fa8b

File: duffy/app/main.py (hunk 2 above)
Comment (gridhead) [reply duffy/app/main.py]: Addressed here https://github.com/CentOS/duffy/pull/133/commits/1c52ef02660e7638be331c5362d113ebb4e6fa8b

File: duffy/app/main.py (hunk 1 above)
Comment (gridhead) [reply duffy/app/main.py]: Addressed here https://github.com/CentOS/duffy/pull/133/commits/1c52ef02660e7638be331c5362d113ebb4e6fa8b

**Review by nphilipp — CHANGES_REQUESTED** [review]
It just struck me that this one and #138 will conflict, so we should rebase this PR on top of #138 and follow suit with the project ⇒ tenant rename.

File: duffy/app/main.py [hunk 5]
```diff
@@ -2,10 +2,29 @@
...
+
+description = """
+Duffy is the middle layer running [`ci.centos.org`](https://ci.centos.org). It provisions, tears
+down and rebuilds physical and virtual machines are used to run tests in the CentOS CI Cluster.
+"""
+
+tags_metadata = [
+    {"name": "nodes", "description": "Operations with physical and virtual nodes"},
+    {"name": "sessions", "description": "Operations with sessions"},
+    {"name": "tenants", "description": "Operations with tenants"},
```
Comment (nphilipp) [inline duffy/app/main.py:18]: Just noticed that this is missing an entry for `chassis`, would you add it?

**Review by nphilipp — CHANGES_REQUESTED** [review]
Other than the missing tags metadata entry for chassis it looks good to me.

File: duffy/app/main.py (hunk 5 above)
Comment (gridhead) [reply duffy/app/main.py:18]: Done in https://github.com/CentOS/duffy/pull/133/commits/be9652cb91922a8ac5451a81a57be54eae6dee47. Good for another look.

**Review by nphilipp — APPROVED** [review]
LGTM!

**Pull request merged**